File: Creational/Prototype/prototype.cs
using System;

public class Prototype : ICloneable
{
    public string Value { get; set; }

    public Prototype(string value)
    {
        Value = value;
    }

    public object Clone()
    {
        return MemberwiseClone();
    }

    public void ShowInfo()
    {
        Console.WriteLine($"Valor: {Value}");
    }
}

// Ejemplo de uso
class Program
{
    static void Main(string[] args)
    {
        Prototype original = new Prototype("Original");
        Prototype clone = (Prototype)original.Clone();

        Console.WriteLine("Original:");
        original.ShowInfo();

        Console.WriteLine("Clon:");
        clone.ShowInfo();

        // Modificar el clon
        clone.Value = "Modificado";
        Console.WriteLine("Después de modificar el clon:");
        original.ShowInfo();
        clone.ShowInfo();
    }
}

File: Creational/Prototype/prototype.py
import copy

class Prototype:
    def __init__(self, value):
        self.value = value

    def clone(self):
        return copy.deepcopy(self)

    def show_info(self):
        print(f"Valor: {self.value}")

# Ejemplo de uso
if __name__ == "__main__":
    original = Prototype("Original")
    clone = original.clone()

    print("Original:")
    original.show_info()

    print("Clon:")
    clone.show_info()

    # Modificar el clon
    clone.value = "Modificado"
    print("Después de modificar el clon:")
    original.show_info()
    clone.show_info()